dataplicity/m2mmanager.py

```diff
@@ -191,6 +191,8 @@ def decode_string(value):
             self.open_file_service(data["port"], data["path"])
         elif action == "run-command":
             self.open_command_service(data["port"], data["command"])
+        elif action == "scan-directory":
```

Comment:
Doesn't actually need / use m2m. The service just schedules a scan in another thread. Can't do it in the main thread as a scan could take multiple seconds and we don't want to hold up terminals etc.